frontend/pre_analysis.py

```diff
@@ -135,16 +133,18 @@ def analyze_classes(self):
             class_attributes = set()
             class_to_instance_attributes[cls.name] = instance_attributes
             class_to_class_attributes[cls.name] = class_attributes
-            class_funcs = []
+
+            class_funcs = {}
             class_to_funcs[cls.name] = class_funcs
 
             # Inspect all class-level statements
             for cls_stmt in cls.body:
                 if isinstance(cls_stmt, ast.FunctionDef):
+                    class_funcs[cls_stmt.name] = (len(cls_stmt.args.args), [d.id for d in cls_stmt.decorator_list],
```

Comment:
This will raise an exception if there is a decorator that's an attribute (e.g. a property setter), could we reject this a bit more gracefully?